oci/oci.go

```diff
@@ -62,6 +47,67 @@ const (
 	UntrustedRuntime = "untrusted"
 )
 
+// Runtime is the generic structure holding both global and specific
+// information about the runtime through RuntimeBase and RuntimeImpl
+// respectively.
+type Runtime struct {
+	RuntimeBase
+	RuntimeImpl
+}
+
+// RuntimeBase gathers informations that can be shared across Runtime
+// implementations.
+type RuntimeBase struct {
+	name                     string
+	trustedPath              string
+	untrustedPath            string
+	trustLevel               string
+	runtimes                 map[string]RuntimeHandler
+	conmonPath               string
+	conmonEnv                []string
+	cgroupManager            string
+	containerExitsDir        string
+	containerAttachSocketDir string
+	logSizeMax               int64
+	noPivot                  bool
+	ctrStopTimeout           int64
+}
+
+// RuntimeImpl is an interface used by the caller to interact with the
+// container runtime. The purpose of this interface being to abstract
+// implementations and their associated assumptions regarding the way to
+// interact with containers. This will allow for new implementations of
+// this interface, especially useful for the case of VM based container
+// runtimes. Assumptions based on the fact that a container process runs
+// on the host will be limited to the RuntimeV1 implementation.
+type RuntimeImpl interface {
```

Comment:
One suggestion from my side is to add the `ValidateRuntimeHandler` to the interface too. The other small thing is to hide the parameter names from the interface description like this:

```go
DeleteContainer(c *Container) error
```
to

```go
DeleteContainer(*Container) error
```

Reply:
> One suggestion from my side is to add the `ValidateRuntimeHandler` to the interface too.

Well I didn't do it on purpose since this is not specific to the type of runtime. That's why I pushed it under `RuntimeBase`. If I follow your suggestion, I'll have to duplicate the same implementation for every single RuntimeImpl implementation.

> The other small thing is to hide the parameter names from the interface description like this.

Sure I can do that :)

Reply:
done